qt/scientific_interfaces/Inelastic/Analysis/FqFitDataView.cpp

```diff
@@ -25,13 +27,45 @@ QStringList FqFitHeaders() {
 
 namespace MantidQt::CustomInterfaces::IDA {
 
-FqFitDataView::FqFitDataView(QWidget *parent) : FqFitDataView(FqFitHeaders(), parent) {}
+FqFitDataView::FqFitDataView(QWidget *parent) : FqFitDataView(FqFitHeaders(), parent) {
+  connect(m_uiForm->pbAdd, SIGNAL(clicked()), this, SLOT(notifyAddClicked()));
+}
 
 FqFitDataView::FqFitDataView(const QStringList &headers, QWidget *parent) : IndirectFitDataView(headers, parent) {
   auto header = m_uiForm->tbFitData->horizontalHeader();
   header->setSectionResizeMode(1, QHeaderView::Stretch);
 }
 
+IAddWorkspaceDialog *FqFitDataView::getAddWorkspaceDialog() {
+  m_addWorkspaceDialog = new FqFitAddWorkspaceDialog(parentWidget());
+
+  connect(m_addWorkspaceDialog, SIGNAL(addData()), this, SLOT(notifyAddData()));
+  connect(m_addWorkspaceDialog, SIGNAL(workspaceChanged(FqFitAddWorkspaceDialog *, const std::string &)), this,
+          SLOT(notifyWorkspaceChanged(FqFitAddWorkspaceDialog *, const std::string &)));
+  connect(m_addWorkspaceDialog, SIGNAL(parameterTypeChanged(FqFitAddWorkspaceDialog *, const std::string &)), this,
+          SLOT(notifyParameterTypeChanged(FqFitAddWorkspaceDialog *, const std::string &)));
+
+  return m_addWorkspaceDialog;
+}
+
+void FqFitDataView::notifyAddClicked() {
+  if (auto presenter = dynamic_cast<FqFitDataPresenter *>(m_presenter)) {
```

Comment:
Yes, it casts `m_presenter` to a `FqFitDataPresenter *` from a `IndirectFitDataPresenter *` and assigns the result to the `presenter` variable. The if statement then checks that `presenter` is not a nullptr (which it would be if the cast fails).

Its necessary to cast from `IndirectFitDataPresenter` to `FqFitDataPresenter` because the later has a few extra member functions which are otherwise inaccessible unless you do the cast. I guess the if statement isn't technically necessary in this case because we should always expect the `FqFitDataView` to correspond to a `FqFitDataPresenter`, so the dynamic cast should never return a nullptr unless a developer has made a mistake.

Reply:
I see, thank you so much for the explanation!